### DISCUSSION TITLE
feature request - clip rendering options

### DISCUSSION THREAD

**Discussion by WeissShahaf** [talmolab/sleap#1455] [Ideas]
Dear Devs,
i am super happy with the sleap pose estimation. 
but i have a few requests for the clip rendering output. nothing major just some esthetic / UX issues:

1. Include rendering options in the GUI: node sizes, color (it's always blue big dots). with/without skeleton.

2. it would be very helpful if there was an option to have the color of dots / instance reflect their score.  this is one i've heard from other colleagues as well.

Thank you in advance,
Shahaf

**Comment by gitttt-1234** [comment]
Closing this discussion, the rendering improvements are now available through [`sleap-io`](https://github.com/talmolab/sleap-io)'s `render_video()` function, which has replaced the older rendering pipeline:

```python
import sleap_io as sio

labels = sio.load_file("predictions.slp")
sio.render_video(
    labels,
    "output.mp4",
    video=labels.videos[0],
    fps=30,
    color_by="track",       # or "instance", "node", "auto"
    palette="tableau10",     # multiple palette options
    marker_size=4,           # configurable node size
    show_edges=True,         # toggle skeleton edges
    line_width=2,            # configurable edge width
    marker_shape="circle",   # circle, square, diamond, triangle
    preset="final",          # quality: "preview", "draft", "final"
)
```

The SLEAP GUI integration was also updated in [PR #2558](https://github.com/talmolab/sleap/pull/2558) with live preview and non-blocking progress.
